src/Engine/ProtoCore/Properties/Resources.resx

```diff
@@ -160,37 +160,37 @@
     <value>Converting an array to {0} would cause array rank reduction and is not permitted.</value>
   </data>
   <data name="kConvertDoubleToInt" xml:space="preserve">
-    <value>Converting double to int will cause possible information loss.</value>
+    <value>Converting a double to an int may lose information.</value>
   </data>
   <data name="kConvertNonConvertibleTypes" xml:space="preserve">
-    <value>Asked to convert non-convertible types.</value>
+    <value>Asked to convert between non-convertible types.</value>
   </data>
   <data name="kCyclicDependency" xml:space="preserve">
-    <value>Cyclic dependency detected at '{0}' and '{1}'.</value>
+    <value>A cyclic dependency was detected at '{0}' and '{1}'.</value>
   </data>
   <data name="kDeferencingNonPointer" xml:space="preserve">
-    <value>Dereferencing a non-pointer.</value>
+    <value>Internal error, please report: Dereferencing a non-pointer. (3f47aacd)</value>
   </data>
   <data name="kExceptionTypeUndefined" xml:space="preserve">
-    <value>Exception type '{0}' is not defined.</value>
+    <value>The exception type '{0}' is not defined.</value>
   </data>
   <data name="kFailToConverToFunction" xml:space="preserve">
-    <value>Converting other things to function pointer is not allowed.</value>
+    <value>Cannot convert objects to function pointers.</value>
   </data>
   <data name="kFailToConverToNull" xml:space="preserve">
-    <value>Converting other things to null is not allowed.</value>
+    <value>Cannot convert objects to null, try assigning null to the variable instead.</value>
   </data>
   <data name="kFailToConverToPointer" xml:space="preserve">
-    <value>Converting other things to pointer is not allowed.</value>
+    <value>Cannot convert objects to pointers.</value>
   </data>
   <data name="kFFIFailedToObtainObject" xml:space="preserve">
-    <value>Failed to obtain object '{0}' for '{1}.{2}'.</value>
+    <value>Internal error, please report: Failed to obtain object '{0}' for '{1}.{2}' (edb6d7d9).</value>
   </data>
   <data name="kFFIFailedToObtainThisObject" xml:space="preserve">
-    <value>Failed to obtain this object for '{0}.{1}'.</value>
+    <value>Internal error, please report: Failed to obtain this object for '{0}.{1}' (88facceb).</value>
   </data>
   <data name="kFFIInvalidCast" xml:space="preserve">
-    <value>'{0}' is being cast to '{1}', but the allowed range is [{2}..{3}].</value>
+    <value>Internal error, please report: '{0}' is being cast to '{1}', but the allowed range is [{2}..{3}]. (dff21092)</value>
```

Comment:
This is just a warning message when input type at C# side is byte or short and it's being assigned with higher integer value.